nsqd/diskqueue.go

```diff
@@ -569,15 +569,15 @@ func (d *diskQueue) handleReadError() {
 func (d *diskQueue) ioLoop() {
 	var dataRead []byte
 	var err error
-	var count int64
+	var rcount int64
+	var wcount int64
 	var r chan []byte
 
 	syncTicker := time.NewTicker(d.syncTimeout)
 
 	for {
 		// dont sync all the time :)
-		if count == d.syncEvery {
-			count = 0
+		if wcount == d.syncEvery {
```

Comment:
don't we want `(wcount + rcount) == d.syncEvery` ? 

And more generally, do we need to distinguish between read and write counts, can we always use a single `count` var?